src/Core/Tool/Authorizer/UserAuthorizer.php

```diff
@@ -86,7 +86,7 @@ public function isAllowed(Entity $entity, $privilege)
 		}
 
 		// Regular user can always read
-		if (in_array($privilege, ['read', 'search'])) {
+		if ($user->getId() and in_array($privilege, ['read', 'search'])) {
```

Comment:
Shouldn't this only use the 		if ($this->acl->hasPermission($user, Permission::MANAGE_USERS)) for search ?

Reply:
and isUserOwner for read